server/service/campaigns.go

```diff
@@ -156,17 +156,17 @@ func (svc *Service) NewDistributedQueryCampaign(ctx context.Context, queryString
 		}
 	}
 
-	err = svc.liveQueryStore.RunQuery(strconv.Itoa(int(campaign.ID)), queryString, hostIDs)
-	if err != nil {
-		return nil, ctxerr.Wrap(ctx, err, "run query")
-	}
-
 	// Metrics are used for total hosts targeted for the activity feed.
 	campaign.Metrics, err = svc.ds.CountHostsInTargets(ctx, filter, targets, time.Now())
 	if err != nil {
 		return nil, ctxerr.Wrap(ctx, err, "counting hosts")
 	}
 
+	err = svc.liveQueryStore.RunQuery(strconv.Itoa(int(campaign.ID)), queryString, hostIDs)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "run query")
+	}
```

Comment:
One of the bugs I reproduced was caused by context being canceled when executing `svc.ds. CountHostsInTargets ` (which meant leaving a live query orphaned in Redis)

PS: Such live query was removed from Redis after running `cleanups_then_aggregation` followed by `frequent_cleanups`. But it's probably not optimal as it can be left orphaned for 1 hour and 15 minutes or so.

Reply:
Not that I can think of. (The swaped operations are independent.)